### PR TITLE
Form add goodeal

### PR DESCRIPTION
**Pull request by cruiser22**
Form to add our "good deal"

### CODE REVIEW

**Review by AlexMouth — APPROVED** [review]

File: src/Controller/AnnouncementController.php [hunk 1]
```diff
@@ -65,4 +65,50 @@ public function index(): string
...
+//Form add annonce
+    public function showFormAddGoodeal(): string
+    {
+
+        {
+            return $this->twig->render('Announcement/addGoodeal.html.twig');
+        }
+
+
+       /* $announcementManager = new AnnouncementManager();
```
Comment (Benja-san) [inline src/Controller/AnnouncementController.php]: kill dead code please

File: src/Controller/FormAddGoodealController.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+use App\Model\AuthorManager;
+
+class FormAddGoodealController extends AbstractController
+{
+    public function index(): string
+    {
+        return $this->twig->render('Announcement/addGoodeal.html.twig');
+    }
+/*
+  private function cleanValue(array $data): array
```
Comment (Benja-san) [inline src/Controller/FormAddGoodealController.php]: clean dead code please

File: src/Model/AddGoodealManager.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        $statement->bindValue(':date_start', $announcement['start-date'], \PDO::PARAM_STR);
+        $statement->bindValue(':date_end', $announcement['end-date'], \PDO::PARAM_STR);
+        $statement->bindValue(':image', $announcement['image'], \PDO::PARAM_STR);
+        $statement->bindValue(':city', $announcement['city'], \PDO::PARAM_STR);
+        $statement->bindValue(':zipcode', $announcement['zipcode'], \PDO::PARAM_INT);
+
+
+        $statement->execute();
+
+/*       $regionId = new RegionManager();
```
Comment (Benja-san) [inline src/Model/AddGoodealManager.php]: clean dead code

File: src/Model/AuthorManager.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    /**
+     * List of regions
+     */
+    public function selectAllAuthor(): array
+    {
+        $statement = $this->pdo->prepare("SELECT * FROM `" . self::TABLE . "`");
+        $statement->execute();
+        return $statement->fetchAll();
+    }
+ /*   public function selectAuthorById(): int
```
Comment (Benja-san) [inline src/Model/AuthorManager.php]: dead code

File: src/Model/RegionManager.php [hunk 5]
```diff
@@ -17,4 +17,10 @@ public function select(): array
         $statement->execute();
         return $statement->fetchAll();
     }
+/*    public function selectRegionById($announcement): int
```
Comment (Benja-san) [inline src/Model/RegionManager.php]: dead code

File: src/View/Announcement/addGoodeal.html.twig [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+	- Ajouter un GooDeal
+{% endblock %}
+
+{% block stylesheet %}
+	<link rel="stylesheet" href="/assets/css/addGoodeal.css">
+{% endblock %}
+
+
+{% block content %}
+	<div class="form">
```
Comment (Benja-san) [inline src/View/Announcement/addGoodeal.html.twig:16]: main tag required

File: src/View/Announcement/addGoodeal.html.twig [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+				<div class="label-column">
+					{{ form.input("lastname","Ton nom*", "text", "lastname", "lastname", errors.lastname, "ex: Juste") }}
+				</div>
+
+				<div class="label-column">
+					{{ form.input("firstname","Ton Prénom*", "text", "firstname", "firstname", errors.firstname, "ex: Axel") }}
+				</div>
+
+				<!-- <label for="category">Choisis ta Catégorie*</label>
```
Comment (Benja-san) [inline src/View/Announcement/addGoodeal.html.twig]: dead coee? or use twig syntax for comments

File: src/View/Announcement/addGoodeal.html.twig [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+																												<select name="category" id="category" required>
+																													<option value="Event">Evenement</option>
+																													<option value="food-service">Restauration</option>
+																													<option value="accommodation">Logement</option>
+																												</select></div><div class="label-column"> -->
+				<div class="label-column">
+					{{ form.input("adress","Nom et Numéro de rue du GooDeal*", "text", "adress", "adress", errors.adress, "ex: 14 bis route nationale") }}
+				</div>
+
+				<!--<label for="region">Région du GooDeal*</label>
```
Comment (Benja-san) [inline src/View/Announcement/addGoodeal.html.twig]: same here

File: src/View/Announcement/addGoodeal.html.twig [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+					{{ form.input("email", "Ton adresse mail*", "email", "email", "email", errors.email, "ex: florent.dupond@gmail.com") }}
+				</div>
+				<button type="submit">Ajouter ton deal</button>
+
+
+				<!-- 			
+																												<div class="label-column">
+																												<label for="start-date">Date de début</label>
+																												<input type="date" id="start-date" name="start-date" placeholder="ex: 20/12/2022"></div><div class="label-column">
+																												<label for="end-date">Date de fin</label>
```
Comment (Benja-san) [inline src/View/Announcement/addGoodeal.html.twig]: big empty space

File: src/Model/AddGoodealManager.php [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        $statement->bindValue(':date_start', $startDate, PDO::PARAM_STR);
+        $statement->bindValue(':date_end', $endDate, PDO::PARAM_STR);
+        $statement->bindValue(':image', $announcement['image'], PDO::PARAM_STR);
+        $statement->bindValue(':city', $announcement['city'], PDO::PARAM_STR);
+        $statement->bindValue(':zipcode', $announcement['zipcode'], PDO::PARAM_INT);
+
+
+        $statement->execute();
+
+/*    public function update(array $item): bool
```
Comment (Benja-san) [inline src/Model/AddGoodealManager.php]: clean dead code

**Review by Benja-san — CHANGES_REQUESTED** [review]

File: src/Controller/AnnouncementController.php (hunk 1 above)
Comment (cruiser22) [reply src/Controller/AnnouncementController.php]: done

File: src/Controller/FormAddGoodealController.php (hunk 2 above)
Comment (cruiser22) [reply src/Controller/FormAddGoodealController.php]: done

File: src/Model/AddGoodealManager.php (hunk 3 above)
Comment (cruiser22) [reply src/Model/AddGoodealManager.php]: ok

File: src/Model/AuthorManager.php (hunk 4 above)
Comment (cruiser22) [reply src/Model/AuthorManager.php]: done

File: src/Model/RegionManager.php (hunk 5 above)
Comment (cruiser22) [reply src/Model/RegionManager.php]: done

File: src/View/Announcement/addGoodeal.html.twig (hunk 6 above)
Comment (cruiser22) [reply src/View/Announcement/addGoodeal.html.twig:16]: done

File: src/View/Announcement/addGoodeal.html.twig (hunk 7 above)
Comment (cruiser22) [reply src/View/Announcement/addGoodeal.html.twig]: done

File: src/View/Announcement/addGoodeal.html.twig (hunk 8 above)
Comment (cruiser22) [reply src/View/Announcement/addGoodeal.html.twig]: done

File: src/View/Announcement/addGoodeal.html.twig (hunk 9 above)
Comment (cruiser22) [reply src/View/Announcement/addGoodeal.html.twig]: done

**Review by Benja-san — APPROVED** [review]

**Pull request merged**